src/gui/widgets/settings.cpp

```diff
@@ -585,14 +585,25 @@ const builder_window::window_resolution& get_window_builder(const std::string& t
 
 	VALIDATE(!resolutions.empty(), formatter() << "Window '" << type << "' has no resolutions.\n");
 
+	const builder_window::window_resolution* best_resolution = nullptr;
+	int best_resolution_score = 0;
+
 	for(const auto& res : resolutions) {
-		if(settings::screen_width <= res.window_width && settings::screen_height <= res.window_height) {
```

Comment:
Uhh, if the window is becoming smaller than 800x600, then that's a bug which needs to be fixed.